packages/devtools_app/lib/src/code_size/code_size_screen.dart

```diff
@@ -98,24 +115,67 @@ class CodeSizeBodyState extends State<CodeSizeBody>
               controller: _tabController,
               tabs: tabs,
             ),
+            Row(
+              children: [
+                if (!isSnapView) _buildDiffTreeTypeDropdown(),
+                const SizedBox(width: defaultSpacing),
+                buildClearButton(isSnapView),
+              ],
+            ),
           ],
         ),
         Expanded(
           child: TabBarView(
             physics: defaultTabBarViewPhysics,
             children: [
-              SnapshotView(),
-              DiffView(),
+              SnapshotView(rootNode: snapshotRoot),
+              DiffView(rootNode: diffRoot),
             ],
             controller: _tabController,
           ),
         ),
       ],
     );
   }
+
+  DropdownButtonHideUnderline _buildDiffTreeTypeDropdown() {
+    return DropdownButtonHideUnderline(
+      child: DropdownButton<DiffTreeType>(
+        value: controller.activeDiffTreeType.value,
+        items: [
+          _buildMenuItem(DiffTreeType.combined),
+          _buildMenuItem(DiffTreeType.increaseOnly),
+          _buildMenuItem(DiffTreeType.decreaseOnly),
+        ],
+        onChanged: diffRoot != null
+            ? (newDiffTreeType) {
+                controller.changeActiveDiffTreeType(newDiffTreeType);
+              }
+            : null,
+      ),
+    );
+  }
+
+  DropdownMenuItem _buildMenuItem(DiffTreeType diffTreeType) {
+    return DropdownMenuItem<DiffTreeType>(
+      value: diffTreeType,
+      child: Text(diffTreeType.display),
+    );
+  }
+
+  Widget buildClearButton(bool isSnapView) {
+    return clearButton(
+      busy: isSnapView ? snapshotRoot == null : diffRoot == null,
+      onPressed: isSnapView ? controller.clearSnapshot : controller.clearDiff,
```

Comment:
can we track which view is active in the controller, and then if you call `controller.clear()`, the controller will clear which ever view needs to be cleared. This will allow us to push the state back down the tree to where it was, and reduce the work of the UI code to manage state.

Reply:
Done